### ISSUE TITLE
[4.0] [backend] com_cache "Purge" view missing

### ISSUE THREAD

**Issue by C-Lodder** [joomla/joomla-cms#25971]
### Steps to reproduce the issue

Go to `/administrator/index.php?option=com_cache&view=purge`

### Expected result

A page

### Actual result

> 404 View not found [name, type, prefix]: purge, html, Administrator

**Comment by brianteeman** [comment]
Relevant PR that broke this https://github.com/joomla/joomla-cms/pull/24077 ??

**Comment by C-Lodder** [comment]
@brianteeman Ah, I think the issue is that I'm using the `Modern` menu preset, whereas the default is set to `Joomla`. I suppose the items simply need removing from the other preset

**Comment by brianteeman** [comment]
the presets havent been touched afaict

**Comment by infograf768** [comment]
We have exactly the same error when we use the joomla preset and click in the System dashboard on the "Clear Expired Cache".
Is there any reason why this view is no more in J4?

**Comment by C-Lodder** [comment]
@infograf768 May have been merged into a single view? Not sure

**Comment by infograf768** [comment]
Hmm, it is not in the cache view and the functionnality was quite different.
The purge method is still present in CacheModel and Controller.

**Comment by infograf768** [comment]
Help on this for 3.x is https://help.joomla.org/proxy?keyref=Help39:Site_Maintenance_Purge_Expired_Cache&lang=en:
```
This tool will purge all expired cache files from the cache folders. Cache files are temporary files that are created to improve the performance of your site. Cache files that are still current according to what Joomla! finds will not be deleted.

    Because this process has to check each cache file individually, it is slower and requires more system resources. Once the purge process is complete there should be no loss in speed for your users, because all up to date cache files are still available for your current Joomla! website.
    In contrast, the Clear Cache option runs more quickly since it deletes all of the files stored in each cache folder. However, the website may be a little slower immediately after running Clean Cache, since all cache files -- including current ones -- are removed, so users lose the benefit of using the cache files until they are re-created by Joomla! to be up to date with your current site.
    If you have a large number of cache files and your server setting for 'max_input_time' is set to less then 60 seconds, you may receive a PHP warning message saying that the process exceeded the maximum time allowed. In this case, no harm is done but all of the files may not have been purged. It is ok simply to re-run the process.
```

**Comment by infograf768** [comment]
code is weird
we have in the unique view
`$toolbar->appendButton('Confirm', 'COM_CACHE_RESOURCE_INTENSIVE_WARNING', 'delete', 'COM_CACHE_PURGE_EXPIRED', 'purge', false);`
@Hackwar Can you look at this please.

**Comment by infograf768** [comment]
OK, found it.
there is a mistake in cache view.html code
we should have
`$toolbar = Toolbar::getInstance('toolbar');`
and not
`$toolbar = Factory::getContainer()->get(ToolbarFactoryInterface::class)->createToolbar('toolbar');`

Thereafter we can see the button
<img width="775" alt="Screen Shot 2019-08-22 at 15 50 22" src="https://user-images.githubusercontent.com/869724/63520197-94431780-c4f4-11e9-9c01-ad3f593f75e7.png">

Therefore we should correct that and take off from the Presets the part concerning the purge link (or change the link to be the same as Clear Cache);

**Comment by 810** [comment]
i still get View not found [name, type, prefix]: purge, html, Administrator

**Comment by infograf768** [comment]
@810
I will make a patch today.

**Comment by infograf768** [comment]
Please test https://github.com/joomla/joomla-cms/pull/26003
closing as we have patch
